Reference/Common-Pitfalls/index.md

```diff
@@ -0,0 +1,466 @@
+by the garbage collector. Request scoped object instances are not accessed by every other thread in the application - __unless you do something like the above!__
+
+An example of an application scoped instance is Umbraco's `ApplicationContext`, this single instance is shared by all threads and exists for the lifetime of
+the application. 
+
+An example of a request scoped instance is the `HttpContext`, this object exists for a single request, it definitely cannot be shared between other threads and especially
+not other request threads because this is where the security information for a given user is stored! The `UmbracoContext` is also a request scoped object - in fact it 
+relies directly on an instance of `HttpContext`. The `UmbracoHelper` is a request scoped object - as you can see above, it requires an instance of an `UmbracoContext`.
+
+So... in the above example, the `UmbracoHelper` which relies on an `UmbracoContext` which relies on an `HttpContext` will now be statically assigned to a variable, which means
+that these particular request scoped objects are now bound to an Application scope lifetime and will never go away. This could mean that under certain circumstances that an entire Umbraco
+cache copy is stuck in memory, or that the `Security` property of the context will be accessed by multiple threads but this now contains the security information for a user for another request!
+
+__Other Examples:__
+
+```csharp
+private static _umbracoContext = UmbracoContext.Current;
+```
+
+```csharp
+//MembershipHelper is also a request scoped instance - it relies either on an UmbracoContext or an HttpContext
+private static _membershipHelper = new MembershipHelper(UmbracoContext.Current);
+```
+
+```csharp
+private static _request = HttpContext.Current.Request;
+```
+
+## Querying with Descendants, DescendantsOrSelf
+
+It's not 100% bad that you use these queries, you just need to understand the implications. 
+Here's a particularly bad scenario:
+
+* You have 10,000 content items in your tree
+* Your tree structure is something like this:
+```
+- Root
+-- Home
+--- Blog (list view with 9495 nodes)
+--- Office Locations (list view with 500 nodes)
+--- About Us
+--- Contact Us
+```
+
+You create a menu on your Home page like:
+
+```
+<ul>
+    <li><a href="@Model.Site().Url">@Model.Site().Name</a></li>
+    @foreach(var node in Model.Site().DescendantsOrSelf().Where(x => x.Level == 2)) 
+    {
+        <li><a href="@node.Url">@node.Name</a></li>
+    }
+</ul>
+```
+
+Which just renders out: _Home, Blog, Office Locations, About Us, Contact Us_
+
+BUT! ...  this is going to perform most horribly. This is going to iterate over every single node in Umbraco, all 10,000 of them. Further more, 
+this means it is going to allocate 10,000 `IPublishedContent` instances in memory just in order to check it's `Level` value. 
+
+This can easily be re-written as:
+
+```
+<ul>
```

Comment:
Just clarified & checked the query builder in the template editor does a sensible suggestion which it does by using children & not descendants. 

All good here :)